### DISCUSSION TITLE
Road book real-time in navigation, interactive in route planning

### DISCUSSION THREAD

**Discussion by devemux86** [devemux86/cruiser#473] [Q&A] [answered]
Road book with the via points and additional information, real time or interactive.

**Navigation**:
- _Updated in real time_: quickly see the distance + time until the next stops.
(available in navigation actions - press direction icon)

**Route planning**:
- _Interactive_: press the via points to see the distance + time from any stop.

<img src="https://github.com/devemux86/cruiser/assets/3484020/674fdfc6-4578-4293-8999-42f88c68e5ea" width="50%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/856f7052-4e7c-4a03-8dbe-1bcf8ad8bc0e" width="50%"/>

**Comment by MotoUKRider** [comment]
This looks good - and I frequently have to squint/stare at the Waypoints list to work out what the distance/timings mean in relation to the overall journey.  I have the same feelings about the yellow "speech bubble" when you tap a Via-Point pin along a round :confused:

Perhaps it's just me, but I feel that the information is there, just not jumping out at me *at a glance*.

**Reply by devemux86** [reply]
Now with the new road book which has meaningful distance / time information for via points and destination,
I wonder if we need similar information about distance / time in the waypoint management screen or bubbles.

Or we could simplify the display of the waypoints to make them easier to understand?

**Reply by MotoUKRider** [reply]
@devemux86 I think having 2 lines of Distance+Timing (plus those arrows) make it a little unclear at a quick glance.  If you're able to style certain portions of text within a line, then something *like* the Road Book screen above may help in the main Waypoints list.

eg: If I'm looking at the Waypoint list, I'd like the Via-Points to jump out at me with their distance+times :boom:

**Reply by devemux86** [reply]
Waypoints list is more of a management screen to quickly reorder or remove waypoints.
Not so much to see their information, also because if we reorder them, then their distance / time is not valid.

That's why if their distance / time can be covered better by the new road book (a read-only display of via points)
then the waypoint list can be simplified to better serve its purpose.

**Comment by 0709wiwiwi** [comment]
Question. Is this information or already active and available?

**Reply by devemux86** [reply]
Road book in navigation and route planning will be available in the next Beta.

**Comment by devemux86** [comment]
Road book: real-time in navigation, interactive in route planning is available for testing in:
- https://github.com/devemux86/cruiser/discussions/482

Navigation: press direction icon.

**Comment by bios4** [comment]
> available in navigation actions - press direction icon

Error report - 3.0.43.b2:
- load a route, start navigation
- click the direction icon > select "Roadbook" tile
- Cruiser immediately closes down (after a few attempts Android suggests to clear the app cache due to errors)

This happens only when my **current location** is a bit **away from the starting point of the route**.
When changing the starting point of the route to my current location, the procedure properly works...

Please check @devemux86

**Reply by devemux86** [reply]
@bios4 thanks for the report, it will be fixed in the next update.

**Comment by devemux86** [comment]
Road book in navigation will also display the arrival times (in your selected time format).

See the updated image in the [1st post](https://github.com/devemux86/cruiser/discussions/473#discussion-6407660).

**Reply by MotoUKRider** [reply]
In your revised screenshot at the top of the page, is the first Road-Book entry of `0:36 PM` an emulator "bug" ?

**Reply by devemux86** [reply]
@MotoUKRider Thanks for the report, the image is from the emulator, I will check it.

**Reply by devemux86** [reply]
@MotoUKRider it will be fixed in the next update.

Interesting this is how Android returns the time 12:36 PM!

**Comment by devemux86** [comment, marked as answer]
Road book real-time in navigation, interactive in route planning is available in [Cruiser 4.0.0](https://play.google.com/store/apps/details?id=gr.talent.cruiser).

**Comment by ramondevic** [comment]
Cool, I really like it, seeing the list of waipoinys (without shapepoints) is what I needed 
:)
